tests/adapter.spec.ts

```diff
@@ -38,7 +38,7 @@ describe("Adapter", () => {
 
   it(`should find a record`, (done) => {
     adapter.create(resource, deadpoolCreateRequest)
-      .concatMap((response: IHttpResponse) => adapter.findOne(resource, new HttpRequest({data: response.data})))
+      .concatMap((response: IHttpResponse) => adapter.findOne(resource, new HttpRequest({criteria: {id: response.data["id"]}})))
```

Comment:
Since you want both modes to work, you should keep a test where id is not passed through the criteria.